### ISSUE TITLE
Wrong x-axis labels and ticks with `type="numeric"`

### ISSUE THREAD

**Issue by bklingen** [dreamRs/apexcharter#48]
The placement (and/or labels) of the x-axis ticks are somewhat off in this plot of a normal distribution with mean 0 and standard deviation 0.1:
```{r}
x <- seq(-0.3, 0.3, 0.01)
df <- data.frame(x=x, y=dnorm(x, mean=0, sd=0.1))
apex(data=df, aes(x,y), type='spline') %>% 
  ax_xaxis(type='numeric', 
           max=0.3,
           range=0.6, 
           tickAmount=6, 
           labels=list(formatter=format_num(".2f"))
  )
```
![image](https://user-images.githubusercontent.com/4784206/108110958-0e107e80-7062-11eb-980a-04ee70efb8cc.png)

I can't quite figure out what's wrong in the code. I expect the upper bound to be 0.3, and ticks at (-0.3, -0.2, -0.1, 0.0, 0.1, 0.2, 0.3), with the plot centered over 0.0. The funny thing is that the x-value in the hover information is correct.

**Comment by pvictor** [comment]
Seems like a bug with float, it doesn't occur if we use `x*100` or without `tickAmount` option.
I'll open an issue in Apexcharts.js repo.

Victor

**Comment by artofstat** [comment]
Great, thanks. Here is a minimal reproducible example to reproduce the bug with `type="numeric"`, where the x-axis labels are off (the x-value in the hover information is correct, though):
```r
df <- data.frame(x=c(1, 2, 3), y=c(0, 1, 0))
apex(data=df, mapping=aes(x=x, y=y), type="line") %>%
  ax_xaxis(type="numeric")
```
![image](https://user-images.githubusercontent.com/42886499/108561044-96458c80-72cb-11eb-8d96-d0ec44d672d4.png)

**Comment by bklingen** [comment]
I tried in a codepen that had **version 3.4.1** of apexchart, and it did give the expected result:

![image](https://user-images.githubusercontent.com/4784206/108580864-7463fe00-72fb-11eb-8be5-e58a714e18f0.png)

![image](https://user-images.githubusercontent.com/4784206/108580978-fe13cb80-72fb-11eb-8689-3b64794b0e06.png)

**Comment by pvictor** [comment]
Thanks,

Here you used `categories` to set x values, in `apex` I use numeric paired values:

```json
"data": [
        {
          "x": 1,
          "y": 0
        },
        {
          "x": 2,
          "y": 1
        },
        {
          "x": 3,
          "y": 0
        }
      ]
```

The bug here seems to appear if option `tickAmount` is not set.

https://codepen.io/pvictor-r/pen/abBLyBX

**Comment by bklingen** [comment]
Sorry if this is not the right lead (I understand very little about these things), but when you change to **version 3.4.1** in your codepen (at least this is what I think I'm doing in settings), the graph does render correctly, i.e., the bug goes away. I don't know how easy it is to update to this version.

![image](https://user-images.githubusercontent.com/4784206/108730436-857a5e00-74f9-11eb-91e9-73d13f5e260a.png)

**Comment by pvictor** [comment]
3.4.1 is actually an old version (17 Feb 2019)

Last version released is 3.23.1, and is the one used in {apexcharter} GitHub (CRAN use 3.22.2)

**Comment by bklingen** [comment]
I believe the issue was fixed on Apex's gihub a few weeks ago. Any chance this will be rolled over soon, so that code like this gives correct results:
```r
df <- data.frame(x=c(1, 2, 3), y=c(0, 1, 0))
apex(data=df, mapping=aes(x=x, y=y), type="line") %>%
  ax_xaxis(type="numeric")
```
Many thanks!
Bernhard

**Comment by pvictor** [comment]
Yes I openned an issue (https://github.com/apexcharts/apexcharts.js/issues/2249) ;)

I was waiting for an official release, however I build manually the JS bundle from Apexcharts repo so you can try.
You may experience some bugs ^^, let me know.

Victor

**Comment by bklingen** [comment]
Thanks, seems to work just fine for the toy example:

![image](https://user-images.githubusercontent.com/4784206/110994093-43994680-8346-11eb-840b-13e71ebbfee6.png)

And if you want to have "nice" ticks, using `tickAmount` works:
```r
df <- data.frame(x=c(1, 2, 3), y=c(0, 1, 0))
apex(data=df, mapping=aes(x=x, y=y), type="line") %>%
  ax_xaxis(type="numeric", tickAmount=2)
```
![image](https://user-images.githubusercontent.com/4784206/110994857-74c64680-8347-11eb-9255-9772573477f6.png)

Only `tickAmount='dataPoints'` doesn't work as advertises on the Apex help: 
*If you have a numeric xaxis xaxis.type = 'numeric', you can specify tickAmount: 'dataPoints' which would make the number of ticks equal to the number of dataPoints in the chart.*
But I don't have a use case for it.

**Comment by bklingen** [comment]
...and the initial example now also shows correct x-axis labels:
```r
x <- seq(-0.3, 0.3, 0.01)
df <- data.frame(x=x, y=dnorm(x, mean=0, sd=0.1))
apex(data=df, aes(x,y), type='line') %>% 
  ax_xaxis(type='numeric', 
           max=0.3,
           range=0.6, 
           tickAmount=6, 
           labels=list(formatter=format_num(".2f"))
  )
```

![image](https://user-images.githubusercontent.com/4784206/110995509-747a7b00-8348-11eb-8ba4-deb509c26815.png)

**Comment by bklingen** [comment]
There is still a small issue, though, if one wants to set a custom x-axis range: It has to do with the precision of the tick labels. Out of the box, the plot is pretty confusing:
```r
x <- seq(-4, 4, 0.01)
df <- data.frame(x=x, y=dnorm(x))
apex(data=df, aes(x,y), type='line') %>% 
  ax_xaxis(type='numeric', 
           max=5,
           range=10, 
           tickAmount=8
           #labels=list(formatter=format_num(".2f"))
  )
```
![image](https://user-images.githubusercontent.com/4784206/110997462-7265eb80-834b-11eb-8156-ce6dc0c7b9ea.png)

One needs to specify the formatting of the tick labels to get a meaningful x-axis:
```r
x <- seq(-4, 4, 0.01)
df <- data.frame(x=x, y=dnorm(x))
apex(data=df, aes(x,y), type='line') %>% 
  ax_xaxis(type='numeric', 
           max=5,
           range=10, 
           tickAmount=8,
           labels=list(formatter=format_num(".2f"))
  )
```
![image](https://user-images.githubusercontent.com/4784206/110997703-c96bc080-834b-11eb-8542-dedba8f9cfdd.png)

I.e., the -1.25 is rounded and displayed as -1, and the -2.50 is rounded and displayed as -3 in the first chart.

**Comment by pvictor** [comment]
Thanks for the tests!
ApexCharts 3.26.0 has been released and included in apexcharter.
I will push to CRAN soon.
